pkg/ingress/controller/openstack/client.go

```diff
@@ -69,11 +70,21 @@ func NewOpenStack(cfg config.Config) (*OpenStack, error) {
 		return nil, fmt.Errorf("failed to find compute v2 endpoint for region %s: %v", cfg.OpenStack.Region, err)
 	}
 
+	// Get barbican service client.
+	var barbican *gophercloud.ServiceClient
+	barbican, err = openstack.NewKeyManagerV1(provider, gophercloud.EndpointOpts{
+		Region: cfg.OpenStack.Region,
+	})
+	if err != nil {
+		log.Warn("Barbican not suppported.")
```

Comment:
I think you should set `barbican = nil` explicit here. `openstack.NewKeyManagerV1` will also return a `ServiceClient`  in case of an error.
See https://github.com/gophercloud/gophercloud/blob/master/openstack/client.go#L465

But at other locations in the code you check for `os.Barbican == nil`.

Reply:
Agree